### PR TITLE
calling the data from an api

### PR DESCRIPTION
**Pull request by Emmanuelaaron**

### CODE REVIEW

File: src/pages/feeds/BootcampFeedsPage.jsx [hunk 1]
```diff
@@ -1,77 +1,14 @@
...
-            }
-        },
-        // Add more courses as needed
-    ]
-};
+import FeedComponent from "./FeedComponent";
+import { useFetchCourses } from "./customHooks/useFetchCourses";
 
 export default function BootcampFeedsPage() {
+    const {courses} = useFetchCourses()
```
Comment (simonpojok) [inline src/pages/feeds/BootcampFeedsPage.jsx:5]: love this ✅

File: src/pages/feeds/BootcampFeedsPage.jsx [hunk 2]
```diff
@@ -1,77 +1,14 @@
...
-                            <p className="text-gray-700">
-                                <strong>Bootcamp: </strong>{course.bootcamp.name}
-                            </p>
-                            <p className="text-gray-700">
-                                <strong>Bootcamp Description: </strong>{course.bootcamp.description}
-                            </p>
-                        </div>
-                    </div>
+                {courses.data.map((course) => (
+                    <FeedComponent course={course} />
```
Comment (simonpojok) [inline src/pages/feeds/BootcampFeedsPage.jsx]: 👍

Comment (simonpojok) [reply src/pages/feeds/BootcampFeedsPage.jsx]: please add key for each item

File: src/pages/feeds/FeedComponent.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                </p>
+                <p className="text-gray-700">
+                    <strong>Bootcamp Description: </strong>{course.bootcamp.description}
+                </p>
+            </div>
+        </div>
+    )
+}
+
+export default FeedComponent
```
Comment (simonpojok) [inline src/pages/feeds/FeedComponent.jsx]: add one at the end

**Review by simonpojok — APPROVED** [review]
LGTM

File: src/pages/feeds/FeedComponent.jsx (hunk 3 above)
Comment (Emmanuelaaron) [reply src/pages/feeds/FeedComponent.jsx]: done

File: src/pages/feeds/BootcampFeedsPage.jsx (hunk 2 above)
Comment (Emmanuelaaron) [reply src/pages/feeds/BootcampFeedsPage.jsx]: done

File: src/pages/feeds/customHooks/useFetchCourses.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { useEffect, useState } from "react"
+import customInstance from "../../../axios_http_client"
+
+export const useFetchCourses = () =>{
+
+    const [courses, setCourses] = useState({ data: [] })
+    const [errorMessage, setErrorMessage] = useState({})
+
+    useEffect(() => {
```
Comment (simonpojok) [inline src/pages/feeds/customHooks/useFetchCourses.js:9]: error page: [forgotpwd](https://reactrouter.com/en/main/start/tutorial#handling-not-found-errors)

Comment (simonpojok) [reply src/pages/feeds/customHooks/useFetchCourses.js:9]: https://reactrouter.com/en/main/route/loader

**Review by simonpojok — COMMENTED** [review]
please see the links

**Review by simonpojok — APPROVED** [review]
This PR is now good, @kcharles52 you can merge it after review

**Pull request merged**